.claude/hooks/gdd-permission-hook.sh

```
@@ -805,6 +853,76 @@ if [[ "$tool_name" == "PowerShell" ]]; then
+    local raw="$1"
+    # Multi-line input must reach the newline deny untouched — `read`
+    # below would silently drop everything past the first line.
+    if [[ "$raw" != *'\'* || "$raw" == *$'\n'* || "$raw" == *$'\r'* ]]; then
+        printf '%s' "$raw"
+        return
+    fi
+    local -a words=()
+    read -r -a words <<< "$raw"
+    local rebuilt="" w core quote changed=0
+    for w in "${words[@]}"; do
+        if [[ "$w" == *'\'* ]]; then
```

Shellcheck (0.11.0)

[info] Want to escape a single quote? echo 'This is how it'\''s done'.

(SC1003)